### PR TITLE
EWSMailSender fix opening threads

### PR DESCRIPTION
**Pull request by jochman**
## Status
Ready

Fixes: https://github.com/demisto/etc/issues/20817

## Description
close threads at the end of the code execution.
matching updated exchangelib.

## Does it break backward compatibility?
   - No

TESTED AND WORKING

### CODE REVIEW

**Comment by glicht** [comment]
@jochman looking at this again. The thread terminate should be done after the close_connections call.

**Comment by glicht** [comment]
@jochman are you following on this?

**Comment by glicht** [comment]
@kirbles19 please review release notes: https://github.com/demisto/content/edit/fix-ewmailsender-threads/Integrations/EWSMailSender/CHANGELOG.md?pr=%2Fdemisto%2Fcontent%2Fpull%2F5140

**Comment by glicht** [comment]
Created branch with test playbook that fails. Test run before fix (with failure) is available here: 

https://circleci.com/gh/demisto/content/36513

Will merge this into this branch in the next commit.

File: Integrations/EWSMailSender/EWSMailSender.py [hunk 1]
```diff
@@ -315,9 +307,21 @@ def main():
...
-        if isinstance(config, Configuration):
-            # The protocol will not kill its threads after use, so kill it manually
-            config.protocol.thread_pool.terminate()
+        try:
+            # we don't want to leave cached connection around as EWS limits the number of connections
+            # in a very aggressive way. 12 seems to be the default limit
+            # see: https://blogs.msdn.microsoft.com/webdav_101/2018/06/02/you-are-doing-too-much-at-one-time-ewsmaxconcurrency-too-many-concurrent-connections-opened/ # noqa
+            close_connections()
+        except Exception as ex:
+            demisto.info("Failed close_connections (shouldn't happen). Ignoring exception: {}".format(ex))
```
Comment (Itay4) [inline Integrations/EWSMailSender/EWSMailSender.py]: should this be `demisto.error` ?

File: TestPlaybooks/playbook-EWS-Mail-Sender-Test.yml [hunk 2]
```diff
@@ -98,12 +107,123 @@ tasks:
...
+      attachIDs:
+        simple: bad
+      attachNames: {}
+      bcc: {}
+      body: {}
+      cc: {}
+      htmlBody: {}
+      replyTo: {}
+      subject:
+        simple: Bad Email That fials
```
Comment (Itay4) [inline TestPlaybooks/playbook-EWS-Mail-Sender-Test.yml]:
```suggestion
        simple: Bad Email That Fails
```

File: TestPlaybooks/playbook-EWS-Mail-Sender-Test.yml [hunk 3]
```diff
@@ -81,8 +91,7 @@ tasks:
       insecure: {}
       method:
         simple: GET
-      password:
-        simple: '{ENCRYPTED}iyMzr5l9JLzg6ThLOlDRxAU7CU3ePLL2BwDWBf3t3wmTZvUbSvo8oPO5vcqeDDHvgB+hx+7G8D7iRQTv/E8UIg=='
```
Comment (Itay4) [inline TestPlaybooks/playbook-EWS-Mail-Sender-Test.yml]: will this work without a password?

File: Integrations/EWSMailSender/EWSMailSender.py (hunk 1 above)
Comment (glicht) [reply Integrations/EWSMailSender/EWSMailSender.py]: We had it before on info. We can change.

**Comment by kirbles19** [comment]
@jochman Done.

**Comment by glicht** [comment]
@jochman @Itay4 add a new more generic approach to use in both mail sender and ews v2.

**Comment by glicht** [comment]
Build is failing on non related test: Phishing v2 Test - Attachment

**Comment by glicht** [comment]
@jochman as you suggested. I've updated the exchangelib_cleanup method to first close connections then threads.

**Pull request merged**